ksqldb-tools/src/test/java/io/confluent/ksql/tools/migrations/commands/ApplyMigrationCommandTest.java

```diff
@@ -0,0 +1,340 @@
+        Instant.ofEpochMilli(1000), ZoneId.systemDefault()));
+
+    // Then:
+    assertThat(result, is(0));
+    final InOrder inOrder = inOrder(ksqlClient);
+    verifyMigratedVersion(inOrder, 1, MigrationState.MIGRATED);
+    verifyMigratedVersion(inOrder, 2, MigrationState.MIGRATED);
+    inOrder.verify(ksqlClient).close();
+    inOrder.verifyNoMoreInteractions();
+  }
+
+  @Test
+  public void shouldNotApplyMigrationIfPreviousNotFinished()
+      throws IOException, ExecutionException, InterruptedException {
+    // Given:
+    command = PARSER.parse("-a");
+    createMigrationFile(1, NAME, migrationsDir, COMMAND);
+    createMigrationFile(2, NAME, migrationsDir, COMMAND);
+    when(versionQueryResult.get()).thenReturn(ImmutableList.of());
+    when(infoQueryResult.get()).thenReturn(ImmutableList.of(createInfoRow(1, NAME, MigrationState.RUNNING)));
+
+    // When:
+    final int result = command.command(config, cfg -> ksqlClient, migrationsDir, Clock.fixed(
+        Instant.ofEpochMilli(1000), ZoneId.systemDefault()));
+
+    // Then:
+    assertThat(result, is(1));
+    final InOrder inOrder = inOrder(ksqlClient);
+    verifyMigratedVersion(inOrder, 1, MigrationState.MIGRATED);
+    inOrder.verify(ksqlClient).close();
+    Mockito.verify(ksqlClient, Mockito.times(1)).executeStatement(COMMAND);
+  }
+
+  @Test
+  public void shouldLogErrorStateIfMigrationFails()
+      throws IOException, ExecutionException, InterruptedException {
+    // Given:
+    command = PARSER.parse("-n");
+    createMigrationFile(1, NAME, migrationsDir, COMMAND);
+    when(versionQueryResult.get()).thenReturn(ImmutableList.of());
+    when(statementResult.get()).thenThrow(new InterruptedException());
+
+    // When:
+    final int result = command.command(config, cfg -> ksqlClient, migrationsDir, Clock.fixed(
+        Instant.ofEpochMilli(1000), ZoneId.systemDefault()));
+
+    // Then:
+    assertThat(result, is(1));
+    final InOrder inOrder = inOrder(ksqlClient);
+    verifyMigratedVersion(inOrder, 1, MigrationState.ERROR);
+    inOrder.verify(ksqlClient).close();
+    inOrder.verifyNoMoreInteractions();
+  }
+
+  @Test
+  public void shouldSkipApplyIfValidateFails()
+      throws IOException, ExecutionException, InterruptedException {
+    // Given:
+    command = PARSER.parse("-n");
+    createMigrationFile(1, NAME, migrationsDir, COMMAND);
+    createMigrationFile(1, "anotherone", migrationsDir, COMMAND);
+    when(versionQueryResult.get()).thenReturn(ImmutableList.of(createVersionRow("1")));
+    when(infoQueryResult.get()).thenReturn(ImmutableList.of(createInfoRow(1, NAME, MigrationState.MIGRATED)));
+
+    // When:
+    final int result = command.command(config, cfg -> ksqlClient, migrationsDir, Clock.fixed(
+        Instant.ofEpochMilli(1000), ZoneId.systemDefault()));
+
+    // Then:
+    assertThat(result, is(1));
+    Mockito.verify(ksqlClient, Mockito.times(3)).executeQuery(any());
+  }
```

Comment:
Verify that `ksqlClient.executeStatement()` and `ksqlClient.insertInto()` were never called.